apps/documents/views.py

```diff
@@ -0,0 +1,231 @@
+import json
+from functools import cache
+
+from django.contrib.auth.decorators import permission_required
+from django.contrib.auth.mixins import PermissionRequiredMixin
+from django.contrib.postgres.search import TrigramSimilarity
+from django.db import transaction
+from django.db.models import Q
+from django.shortcuts import get_object_or_404, redirect
+from django.urls import reverse
+from django.views.decorators.http import require_POST
+from django.views.generic import ListView, TemplateView
+
+from apps.documents.models import Repository, RepositoryType
+from apps.files.models import File
+from apps.teams.decorators import login_and_team_required
+from apps.teams.mixins import LoginAndTeamRequiredMixin
+
+
+class RepositoryHome(LoginAndTeamRequiredMixin, TemplateView):
+    template_name = "documents/repositories.html"
+
+    def get_context_data(self, team_slug: str, tab_name: str, **kwargs):
+        context = {
+            "active_tab": "manage_files",
+            "title": "Manage Files",
+            "tab_name": tab_name,
+            "files_list_url": reverse("documents:files_list", kwargs={"team_slug": team_slug}),
+            "upload_files_url": reverse("documents:upload_files", kwargs={"team_slug": team_slug}),
+            "collections_list_url": reverse("documents:collections_list", kwargs={"team_slug": team_slug}),
+            "new_collection_url": reverse("documents:new_collection", kwargs={"team_slug": team_slug}),
+            "files_count": File.objects.filter(team__slug=team_slug).count(),
+            "collections_count": Repository.objects.filter(
+                team__slug=team_slug, type=RepositoryType.COLLECTION
+            ).count(),
+        }
+        if tab_name == "files":
+            context["collections"] = Repository.objects.filter(
+                team__slug=team_slug, type=RepositoryType.COLLECTION
+            ).all()
+
+        return context
+
+
+class BaseObjectListView(ListView, PermissionRequiredMixin):
+    details_url_name: str
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context["details_url_name"] = self.details_url_name
+        context["tab_name"] = self.tab_name
+        return context
+
+
+class BaseDetailsView(TemplateView, PermissionRequiredMixin):
```

Comment:
Any reason not to extend Django's `DetailView`? Just curious.

Reply:
I forgot that that is a thing. I'll use that to keep thing Django-y